### PR TITLE
PR: Split History module

### PR DESCRIPTION
**Pull request by rlaverde**
- Move history to a module
- Also decouple  the plugin UI in a widget to make it reusable and easy testable.

```
history/
├── confpage.py
├── __init__.py
├── plugin.py
├── tests
│   └── test_history.py
└── widgets.py
```

### CODE REVIEW

**Comment by goanpeca** [comment]
@rlaverde please fix docstrings. They should end in a period and use a single line (tripplw quotes) if the fit

```
"""Something something something."""
```

or 

```
"""
Something something something.

Extra something.
"""
```

**Comment by ccordoba12** [comment]
@rlaverde this PR doesn't contain any renames, so git won't be able to detect the right file when we want to merge changes from 3.x.

So I think you need to redo this PR and use renames instead.

**Comment by ccordoba12** [comment]
By renames, I mean `git mv`

**Comment by rlaverde** [comment]
There was only one file, I move it in https://github.com/spyder-ide/spyder/pull/4593/commits/23a0615624ef83e017e55acf4fe59590a8085f87

but after that I move some of it:
to confpage: https://github.com/spyder-ide/spyder/pull/4593/commits/b904374b8c8adfe99e2beb41166a3106e4ce1f2e (as in other plugins modules)
to widgets: https://github.com/spyder-ide/spyder/pull/4593/commits/59b3d19e17d6f81773535bb16d815461793d36c3

because of that github doesn't detect the rename, and changes in the code move it to other files will have to be merged manually

File: spyder/plugins/history/widgets.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from qtpy.QtWidgets import (QHBoxLayout, QMenu, QWidget, QToolButton,
+                            QVBoxLayout)
+
+# Local imports
+from spyder.utils import encoding
+from spyder.py3compat import is_text_string, to_text_string
+from spyder.widgets.sourcecode import codeeditor
+from spyder.config.base import _
+from spyder.utils import icon_manager as ima
+from spyder.utils.qthelpers import (add_actions, create_toolbutton)
```
Comment (ccordoba12) [inline spyder/plugins/history/widgets.py]: Parenthesis are not needed here.

**Comment by ccordoba12** [comment]
I left a minor comment, then I'll merge.

**Pull request merged**